src/client/Microsoft.Identity.Client/Http/Headers/WwwAuthenticateParameters.cs

```diff
@@ -0,0 +1,464 @@
+        /// </summary>
+        [Obsolete("The client apps should know which scopes to request for.", true)]
+        public IEnumerable<string> Scopes { get; set; }
+
+        /// <summary>
+        /// Authority from which to request an access token.
+        /// </summary>
+        public string Authority { get; set; }
+
+        /// <summary>
+        /// Claims demanded by the web API.
+        /// </summary>
+        public string Claims { get; set; }
+
+        /// <summary>
+        /// Error.
+        /// </summary>
+        public string Error { get; set; }
+
+        /// <summary>
+        /// Error indicating that parsing failed.
+        /// </summary>
+        public bool ParsingError { get; private set; }
+
+        /// <summary>
+        /// AuthScheme.
+        /// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate#syntax for more details
+        /// </summary>
+        public string AuthScheme { get; set; } 
+
+        /// <summary>
+        /// Server Nonce.
+        /// </summary>
+        public string ServerNonce { get; set; }
+
+        /// <summary>
+        /// Return the <see cref="RawParameters"/> of key <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">Name of the raw parameter to retrieve.</param>
+        /// <returns>The raw parameter if it exists,
+        /// or throws a <see cref="System.Collections.Generic.KeyNotFoundException"/> otherwise.
+        /// </returns>
+        public string this[string key]
+        {
+            get
+            {
+                return RawParameters[key];
+            }
+        }
+
+        /// <summary>
+        /// Dictionary of raw parameters in the WWW-Authenticate header (extracted from the WWW-Authenticate header
+        /// string value, without any processing). This allows support for APIs which are not mappable easily to the standard
+        /// or framework specific (Microsoft.Identity.Model, Microsoft.Identity.Web).
+        /// </summary>
+        internal IDictionary<string, string> RawParameters { get; private set; }
+
+        /// <summary>
+        /// Gets Azure AD tenant ID.
+        /// </summary>
+        public string GetTenantId() => Instance.Authority
+                                               .CreateAuthority(Authority, validateAuthority: true)
+                                               .TenantId;
+        #region Obsolete Api
+        /// <summary>
+        /// Create the authenticate parameters by attempting to call the resource unauthenticated, and analyzing the response.
+        /// </summary>
+        /// <param name="resourceUri">URI of the resource.</param>
+        /// <returns>WWW-Authenticate Parameters extracted from response to the unauthenticated call.</returns>
+        [System.ComponentModel.EditorBrowsable(EditorBrowsableState.Never)]
+        public static Task<WwwAuthenticateParameters> CreateFromResourceResponseAsync(string resourceUri)
```

Comment:
I feel that we should add an "Obsolete" attribute to these ones, because they are not really usable as they are.

Reply:
Some solutions set the use of obsolete methods as an error in their ruleset, which is technically a breaking change. I believe the recommended SDL ruleset has this which can cause issues. Also, the obsolete methods still work, they just don't account for Pop/Multiple schemes. Ill add `<remarks>`